### PR TITLE
Use 1ES runner to build and publish images

### PR DESCRIPTION
**Pull request by wedaly**
Run workflow only on tag "v*" push, not workflow_dispatch.
Remove references to changelog GH action.
Run build and publish steps on 1ES GitHub runners.

### CODE REVIEW

File: .github/workflows/build-publish-mcr.yml [hunk 1]
```diff
@@ -37,33 +40,21 @@ jobs:
             -E unused
             -E varcheck
   build:
-    runs-on: ubuntu-latest
+    runs-on:
+      labels: [self-hosted, "1ES.Pool=1es-aks-ip-masq-agent-pool-ubuntu"]
```
Comment (paulgmiller) [inline .github/workflows/build-publish-mcr.yml:44]: do we have a tsg on who to go to when these break/run out of capacity?

File: .github/workflows/build-publish-mcr.yml [hunk 2]
```diff
@@ -37,33 +40,21 @@ jobs:
...
-      - name: Get Changelog Entry
-        id: changelog_reader
-        uses: mindsers/changelog-reader-action@v2
-        with:
-          validation_depth: 10
-          path: ./CHANGELOG.md
-      - name: Display output
+      - name: Build
         run: |
-           echo "Version: ${{ steps.changelog_reader.outputs.version }}"
```
Comment (paulgmiller) [inline .github/workflows/build-publish-mcr.yml:55]: assume nobody was looking at these?

File: .github/workflows/build-publish-mcr.yml [hunk 3]
```diff
@@ -37,33 +40,21 @@ jobs:
...
-        with:
-          client-id: ${{ secrets.AZURE_CLIENT_ID }}
-          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
-          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
-      - name: 'Run Azure CLI commands'
+          make build
+      - name: Publish
         run: |
+          echo "Logging in with Azure CLI"
+          az login --identity
```
Comment (paulgmiller) [inline .github/workflows/build-publish-mcr.yml:54]: so anybody on this 1espool can push to our repo is that true/fair?

**Review by paulgmiller — APPROVED** [review]

File: .github/workflows/build-publish-mcr.yml [hunk 4]
```diff
@@ -1,5 +1,8 @@
 name: Building and pushing mcr
-on: [workflow_dispatch]
+on:
+  push:
+    tags:
+      - "v*"
```
[the remark this answers is not in the document]
Comment (wedaly) [reply .github/workflows/build-publish-mcr.yml:5]: I updated the repo security settings so only admins can push tags that match "v*"

File: .github/workflows/build-publish-mcr.yml (hunk 2 above)
Comment (wedaly) [reply .github/workflows/build-publish-mcr.yml:55]: there was always a manual step to copy the CHANGELOG into the GitHub release, so I don't think we're losing much by removing this step.

File: .github/workflows/build-publish-mcr.yml (hunk 3 above)
Comment (wedaly) [reply .github/workflows/build-publish-mcr.yml:54]: yes

File: .github/workflows/build-publish-mcr.yml (hunk 1 above)
Comment (wedaly) [reply .github/workflows/build-publish-mcr.yml:44]: I'll update the internal wiki with those steps.

**Pull request merged**